core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -301,6 +298,15 @@ private[spark] class MesosSchedulerBackend(
     }
   }
 
+  def isOfferValidForScheduling(cpusOffered: Double, memory : Double,
+                                slaveId: String, sc : SparkContext): Boolean = {
+    val meetsMemoryRequirements = memory >= calculateTotalMemory(sc)
```

Comment:
Why do you have these variables in fine grain but not in coarse grain?
Let's make them consisitent